Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -479,12 +479,10 @@ namespace Babylon
 
                     m_sessionState = std::make_unique<SessionState>(graphicsImpl);
 
-                    if (!m_system.IsInitialized())
+                    if (!m_system.IsInitialized() &&
+                        !m_system.TryInitialize())
                     {
-                        while (!m_system.TryInitialize())
-                        {
-                            // do nothing
-                        }
+                        return arcana::task_from_error<void>(std::make_exception_ptr(std::runtime_error{"Failed to initialize xr system."})); 
```

Comment:
```suggestion
                        throw std::runtime_error{"Failed to initialize xr system."};
```

Reply:
@ryantrem with these arcana tasks if we are trying to reject the promise, is it better to return the task_from_error or just throw here?

Reply:
@bghgary was this commit suggestion related to not knowing how the api works? or is this the correct way to reject this requestSession promise?

Reply:
A throw here will be converted to a task_from_error automatically in arcana.